tequilapi/contract/identity.go

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (C) 2020 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package contract
+
+import (
+	"github.com/mysteriumnetwork/node/identity"
+	"github.com/mysteriumnetwork/node/tequilapi/validation"
+)
+
+// IdentityDTO holds identity address.
+// swagger:model IdentityDTO
+type IdentityDTO struct {
+	// identity in Ethereum address format
+	// required: true
+	// example: 0x0000000000000000000000000000000000000001
+	Address            string `json:"id"`
+	RegistrationStatus string `json:"registrationStatus,omitempty"`
+	ChannelAddress     string `json:"channelAddress,omitempty"`
+	Balance            uint64 `json:"balance,omitempty"`
+	BalanceEstimate    uint64 `json:"balanceEstimate,omitempty"`
+}
+
+// NewIdentityDTO maps to API identity.
+func NewIdentityDTO(id identity.Identity) IdentityDTO {
+	return IdentityDTO{Address: id.Address}
+}
+
+// ListIdentitiesResponse holds list of identities.
+// swagger:model ListIdentitiesResponse
+type ListIdentitiesResponse struct {
+	Identities []IdentityDTO `json:"identities"`
+}
+
+// NewIdentityListResponse maps to API identity list.
+func NewIdentityListResponse(ids []identity.Identity) ListIdentitiesResponse {
+	result := ListIdentitiesResponse{
+		Identities: make([]IdentityDTO, len(ids)),
+	}
+	for i, id := range ids {
+		result.Identities[i] = NewIdentityDTO(id)
+	}
+	return result
+}
+
+// IdentityGetCurrentRequest request to get/create an identity.
+// swagger:model CurrentIdentityDTO
+type IdentityGetCurrentRequest struct {
```

Comment:
`IdentityCreateRequest`, `IdentityUnlockRequest` and `IdentityGetCurrentRequest` are the same, do we need all of them?

Reply:
Merged into `IdentityRequest`. Thanks!